src/wagtail_ai/views.py

```diff
@@ -98,14 +106,70 @@ def process(request) -> JsonResponse:
     try:
         response = handler(prompt=prompt, text=prompt_form.cleaned_data["text"])
     except AIHandlerException as e:
-        return JsonResponse({"error": str(e)}, status=400)
+        return ErrorJsonResponse(str(e), status=400)
     except Exception:
         logger.exception("An unexpected error occurred.")
-        return JsonResponse({"error": _("An unexpected error occurred.")}, status=500)
+        return ErrorJsonResponse(_("An unexpected error occurred."))
 
     return JsonResponse({"message": response})
 
 
+def user_has_permission_for_image(user, image):
+    from wagtail.images.permissions import permission_policy
+
+    return permission_policy.user_has_permission_for_instance(user, "choose", image)
+
+
+def describe_image(request) -> JsonResponse:
+    form = DescribeImageApiForm(request.POST)
+    if not form.is_valid():
+        return ErrorJsonResponse(form.errors_for_json_response(), status=400)
+
+    model = cast(Type[AbstractImage], get_image_model())
+    image = get_object_or_404(model, pk=form.cleaned_data["image_id"])
+
+    if not user_has_permission_for_image(request.user, image):
+        return ErrorJsonResponse("Access denied", status=403)
+
+    try:
+        backend = ai.get_backend(BackendFeature.IMAGE_DESCRIPTION)
+    except ai.BackendNotFound:
+        return ErrorJsonResponse(
+            "No backend is configured for image description. Please set"
+            " `IMAGE_DESCRIPTION_BACKEND` in `settings.WAGTAIL_AI`.",
+            status=400,
+        )
+
+    rendition = image.get_rendition("max-800x600")
```

Comment:
**Suggestion:** This could be a setting in the future of what rendition size is sent.